s3tests_boto3/functional/test_iam.py

```diff
@@ -0,0 +1,934 @@
+                      UserName=get_tenant_user_id())
+    status = _get_status(e.response)
+    eq(status, 404)
+    response = client.delete_user_policy(PolicyName='AllowAccessPolicy',
+                                         UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+@attr(resource='user-policy')
+@attr(method='delete')
+@attr(operation='Verify Delete multiple User policies for a user')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_delete_user_policy_from_multiple_policies():
+    client = get_iam_client()
+
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": "*",
+             "Resource": "*"}}
+    )
+
+    response = client.put_user_policy(PolicyDocument=policy_document_allow,
+                                      PolicyName='AllowAccessPolicy1',
+                                      UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = client.put_user_policy(PolicyDocument=policy_document_allow,
+                                      PolicyName='AllowAccessPolicy2',
+                                      UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = client.put_user_policy(PolicyDocument=policy_document_allow,
+                                      PolicyName='AllowAccessPolicy3',
+                                      UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = client.delete_user_policy(PolicyName='AllowAccessPolicy1',
+                                         UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = client.delete_user_policy(PolicyName='AllowAccessPolicy2',
+                                         UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = client.get_user_policy(PolicyName='AllowAccessPolicy3',
+                                      UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    eq(response['PolicyName'], "AllowAccessPolicy3")
+    eq(response['PolicyDocument'], json.loads(policy_document_allow))
+
+    response = client.delete_user_policy(PolicyName='AllowAccessPolicy3',
+                                         UserName=get_tenant_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow Bucket Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_allow_bucket_actions_in_user_policy():
+    client = get_iam_client()
+    s3_client = get_tenant_client()
+    bucket = get_new_bucket(client=s3_client)
```

Comment:
here get_new_bucket() gets bucket for the tenant client and then ListBuckets and DeleteBucket is being tested. These actions will be allowed even without the user policy since tenant client is the owner of the bucket.
The bucket must be created for another user in the same tenant and then the policy must be attached to the tenant user and then the policy must be exercised to see whether the tenant user has ListAllMyBuckets and DeleteBucket permission over buckets belonging to the other user.

Reply:
Or if you want to work with the bucket owned by the tenant user, you can modify the policy to only allow Read Access to the bucket and the objects in it. And check if write to the bucket is being 'Denied' correctly.

Reply:
Yes, you are right.
we will do the changes.